test/images/sample-apiserver/Dockerfile

```diff
@@ -0,0 +1,77 @@
+# Copyright 2018 The Kubernetes Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+FROM BASEIMAGE AS build_k8s_1_7_sample_apiserver
```

Comment:
we can use a manifest image here and avoid using the BASEIMAGE logic

Reply:
I don't cross building will really work here, because for `utils.sh` script will build the dockerimage for every supported arch on the native machine(amd64). Until and unless you have qemu binary in it, it will not build on the native arch.

At present all our cross-building logic is only for building images for all the supported archs on `amd64` arch.

Reply:
This logic may work in mac or linux with newer version of kernels(many distros not yet adopted) but will not work on any LTS version of linux AFAIK.